src/spatial_reaction_systems/utility.jl

```diff
@@ -0,0 +1,191 @@
+vals_to_dict(syms::Vector, vals::Vector{<:Vector}) = Dict(zip(syms, vals))
+# Produces a dictionary with all parameter values.
+function param_dict(vert_ps, edge_ps, lrs)
+    merge(vals_to_dict(vertex_parameters(lrs), vert_ps),
+          vals_to_dict(edge_parameters(lrs), edge_ps))
+end
+
+# Computes the transport rates and stores them in a desired format (a Dictionary from species index to rates across all edges).
+function compute_all_transport_rates(vert_ps::Vector{Vector{Float64}}, edge_ps::Vector{Vector{Float64}}, lrs::LatticeReactionSystem)
+    param_value_dict = param_dict(vert_ps, edge_ps, lrs)    # Creates a dict, allowing us to access the values of wll parameters.
+    unsorted_rates = [s => Symbolics.value.(compute_transport_rates(get_transport_rate_law(s, lrs), param_value_dict, lrs.num_edges)) for s in spatial_species(lrs)] # For all species with transportation, compute their transportation rate (across all edges). This is a vector, pairing each species to these rates.
+    return sort(unsorted_rates; by=rate -> findfirst(isequal(rate[1]), species(lrs)))   # Sorts all the species => rate pairs according to their species index in species(::ReactionSystem),
+end
+# For a species, retrieves the symbolic expression for its transportation rate (likely only a single parameter, such as `D`, but could be e.g. L*D, where L and D are parameters).
+function get_transport_rate_law(s::BasicSymbolic{Real}, lrs::LatticeReactionSystem)
+    rates = filter(sr -> isequal(s, sr.species), lrs.spatial_reactions)
+    (length(rates) > 1) && error("Species $s have more than one diffusion reaction.")    # We could allows several and simply sum them though, easy change.
+    return rates[1].rate
+end
+# For the numeric expression describing the rate of transport (likely only a single parameter, such as `D`), and the values of all our parameters, computes the transport rate(s). If all parameters the rate depend on are uniform all edges, this becomes a length 1 vector. Else a vector with each value corresponding to the rate at one specific edge.
+function compute_transport_rates(rate_law::Num,
+                                 param_value_dict::Dict{SymbolicUtils.BasicSymbolic{Real}, Vector{Float64}}, num_edges::Int64)
+    relevant_parameters = Symbolics.get_variables(rate_law)                 # Extracts the parameters that this rate depend on.
+    if all(length(param_value_dict[P]) == 1 for P in relevant_parameters)   # If all these parameters are spatially uniform.
+        return [
+            substitute(rate_law,
+                       Dict(p => param_value_dict[p][1] for p in relevant_parameters)), # Computes a length 1 vector with this value.
+        ]
+    end
+    return [substitute(rate_law,                                            # If at least on parameter the rate depends on have a value varying across all edges, we have to compute one rate value for each edge.
```

Comment:
Not sure how bad it would get on large graphs (especially since it will at worst likely be something like L*D). However, building the function seems strictly better.

Would probably have to play around some with it to get it right. We can either wait with this one until I have figured it out, or add to the missing stuff for spatial issue.